providers/ms365/resources/microsoft.go

```diff
@@ -10,6 +10,41 @@ import (
 	"go.mondoo.com/cnquery/v11/providers/ms365/connection"
 )
 
+type mqlMicrosoftInternal struct {
+	// index users by id
+	idxUsersById map[string]*mqlMicrosoftUser
+	// index users by principal name
+	idxUsersByPrincipalName map[string]*mqlMicrosoftUser
```

Comment:
nit: it doesnt seem we use this index to fetch users, can we remove it if thats the case?

Reply:
Yes, lets remove it